nexus/src/app/multicast/mod.rs

```diff
@@ -0,0 +1,555 @@
+            .await?;
+
+        // Ensure group is in "Active" state (should have `underlay_group_id`)
+        if current_group.state != db::model::MulticastGroupState::Active {
+            return Err(Error::invalid_request(&format!(
+                "cannot update multicast group in state: {state}. group must be in \"Active\" state.",
+                state = current_group.state
+            )));
+        }
+
+        let underlay_group_id =
+            current_group.underlay_group_id.ok_or_else(|| {
+                Error::internal_error(
+                    "active multicast group missing `underlay_group_id`",
+                )
+            })?;
+
+        // Store old name for saga rollback
+        let old_name = current_group.name().clone();
+        // store the old sources
+        let old_sources = current_group.source_ips.clone();
+        // store the old mvlan to detect changes
+        let old_mvlan = current_group.mvlan;
+
+        // Validate the new source configuration if provided
+        if let Some(ref new_source_ips) = params.source_ips {
+            validate_ssm_configuration(
+                current_group.multicast_ip.ip(),
+                new_source_ips,
+            )?;
+        }
+
+        // Update the database first
+        let result = self
+            .db_datastore
+            .multicast_group_update(
+                opctx,
+                MulticastGroupUuid::from_untyped_uuid(group_id.id()),
+                params,
+            )
+            .await?;
+
+        // If name, sources, or mvlan changed, execute DPD update saga to keep
+        // dataplane configuration in sync with the database (including tag updates)
+        if Self::needs_dataplane_update(
+            old_name.as_str(),
+            &params.identity.name,
+            &params.source_ips,
+            old_mvlan,
+            &params.mvlan,
+        ) {
+            let new_name = params
+                .identity
+                .name
+                .as_ref()
+                .map(|n| n.as_str())
+                .unwrap_or(old_name.as_str());
+
+            let saga_params = Params {
+                serialized_authn: Serialized::for_opctx(opctx),
+                external_group_id: current_group.id(),
+                underlay_group_id,
+                old_name: old_name.to_string(),
+                new_name: new_name.to_string(),
+                old_sources,
+                new_sources: params
+                    .source_ips
+                    .as_ref()
+                    .map(|ips| ips.iter().map(|ip| (*ip).into()).collect())
+                    .unwrap_or_else(|| {
+                        // If no source change requested, use current sources from DB
+                        // This is important for SSM groups which require sources
+                        current_group.source_ips.clone()
+                    }),
+            };
+
+            self.sagas.saga_execute::<SagaMulticastGroupDpdUpdate>(saga_params)
```

Comment:
What's the difference between executing this saga and triggering the corresponding multicast RPWs?